lib/x/media_uploader.rb

```diff
@@ -19,9 +19,9 @@ module MediaUploader
     def upload(client:, file_path:, media_category:, media_type: infer_media_type(file_path, media_category),
       boundary: SecureRandom.hex)
       validate!(file_path:, media_category:)
-      upload_body = construct_upload_body(file_path:, media_type:, boundary:)
+      upload_body = construct_upload_body(file_path:, media_type:, media_category:, boundary:)
       headers = {"Content-Type" => "multipart/form-data, boundary=#{boundary}"}
-      client.post("media/upload?media_category=#{media_category}", upload_body, headers:)
+      client.post("media/upload", upload_body, headers:)&.fetch("data")
```

Comment:
Now data is wrapped in `data` attribute so, as for other endpoints, this attribute is returned here.